src/Components/MainMenu/MainMenu.js

```diff
@@ -111,6 +116,18 @@ class MainMenu extends React.Component {
     );
   }
 
+  constructor(props) {
+    super(props);
+    const { menu } = this.props;
+    this.isNested = MainMenu.isDeeplyNested(menu);
+  }
+
+  static isDeeplyNested(menu) {
+    return menu.some(item =>
+      (item.items || []).some(obj => obj.hasOwnProperty('items')),
```

Comment:
as I think about this I'm not sure `obj.hasOwnProperty` is correct. you could get an empty array or a null value and that would produce a nest menu but it would be marked as nested.